command/agent_test.go

```diff
@@ -489,6 +489,124 @@ listener "tcp" {
 	}
 }
 
+// TestAgent_NoAutoAuthTokenIfNotConfigured tests that API proxy will not use the auto-auth token
+// unless configured to.
+func TestAgent_NoAutoAuthTokenIfNotConfigured(t *testing.T) {
+	logger := logging.NewVaultLogger(hclog.Trace)
+	cluster := vault.NewTestCluster(t, nil, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	serverClient := cluster.Cores[0].Client
+
+	// Unset the environment variable so that proxy picks up the right test
+	// cluster address
+	defer os.Setenv(api.EnvVaultAddress, os.Getenv(api.EnvVaultAddress))
+	os.Unsetenv(api.EnvVaultAddress)
+
+	// Create token file
+	tokenFileName := makeTempFile(t, "token-file", serverClient.Token())
+	defer os.Remove(tokenFileName)
+
+	sinkf, err := os.CreateTemp("", "sink.test.")
```

Comment:
Nit: If you use `t.TempDir()` to create this file you won't need to worry about deferring the removal.

Reply:
Great suggestion! I refactored to use `sinkFileName := makeTempFile(t, "sink-file", "")` and it saved seven lines :)

Reply:
...I got a bit ahead of myself and replaced it everywhere in the Agent/Proxy tests :)